src/modules/NfsOptions.rb

```diff
@@ -169,40 +172,136 @@ def validate(options)
+      else
+        option_list << "nfsvers=#{version.mntops_value}"
+      end
 
       to_string(option_list)
     end
 
+    # Whether the given mount options correspond to a NFSv4.1 mount
+    #
+    # This checks the usage of 'nfsvers' and 'vers'. 'minorversion' is ignored
+    # since it should not be used nowadays.
+    #
+    # @param options [String] mount options in the comma-separated format used
+    #   by mount and /etc/fstab
+    # @return [Boolean] is version >= 4.1 enabled
+    def get_nfs41(options)
+      nfs_version(options).mntops_value == "4.1"
+    end
+
+    # Modifies the mount options to make sure NFSv4.1 is used (or to make sure
+    # it stops being used).
+    #
+    # This uses 'nfsvers' (or 'vers' if it was already present). Any
+    # 'minorversion' option will be deleted, since they should not be longer
+    # used.
+    #
+    # @param options [String] mount options in the comma-separated format used
+    #   by mount and /etc/fstab
+    # @param nfs41 [Boolean] whether to enable version >= 4.1. If false the
+    #   options will be configured to use any NFS version.
+    # @return [String] new fstab option string
+    def set_nfs41(options, nfs41)
+      version_string = nfs41 ? "4.1" : nil
+      version = Y2NfsClient::NfsVersion.for_mntops_value(version_string)
+      set_nfs_version(options, version)
+    end
+
+    # Checks whether some of the old options that used to work to configure
+    # the NFS version (but do not longer work now) is used.
+    #
+    # Basically, this checks for the presence of minorversion
+    #
+    # @param options [String] mount options in the comma-separated format used
+    #   by mount and /etc/fstab
+    # @return [Boolean]
+    def legacy?(options)
+      option_list = from_string(options)
+      option_list.any? { |opt| opt.start_with?("minorversion=") }
+    end
+
     publish function: :validate, type: "string (string)"
     publish function: :get_nfs41, type: "boolean (string)"
     publish function: :set_nfs41, type: "string (string, boolean)"
+
+  private
+
+    # Option used to set the NFS protocol version
+    #
+    # @param option_list [Array<String>]
+    # @return [String, nil] contains the whole 'option=value' string
+    def relevant_version_option(option_list)
+      # According to manual tests and documentation, none of the forms has higher precedence.
+      # Use #reverse because in case of conflicting options, the latest one is used by mount
+      option_list.reverse.find do |opt|
+        version_option?(opt)
+      end
+    end
+
+    # Value part for {#relevant_version_option}
+    #
+    # @param option_list [Array<String>]
+    # @return [String, nil]
+    def relevant_version_value(option_list)
+      relevant_option = relevant_version_option(option_list)
+      return nil unless relevant_option
+      parts = relevant_option.split("=")
+      return nil if parts.size == 1
+      parts.last
```

Comment:
and when you use part limited split then code can look like:
```
  key, value = relevant_option.split("=", 2)
  value
```

Reply:
oh, I though I was checking `size != 2`